cpp/collections_module/algorithm/collections.cpp

```diff
@@ -50,59 +83,56 @@ void Collections::Avg(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *res
     }
     average /= list.Size();
 
-    auto record = record_factory.NewRecord();
-    record.Insert(std::string(kResultAvg).c_str(), average);
+    result.SetValue(average);
 
   } catch (const std::exception &e) {
-    record_factory.SetErrorMessage(e.what());
+    result.SetErrorMessage(e.what());
     return;
   }
 }
 
-void Collections::ContainsAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+void Collections::ContainsAll(mgp_list *args, mgp_func_context *ctx, mgp_func_result *res, mgp_memory *memory) {
   mgp::memory = memory;
   const auto arguments = mgp::List(args);
-  const auto record_factory = mgp::RecordFactory(result);
+  auto result = mgp::Result(res);
 
   try {
     const auto list1{arguments[0].ValueList()};
     std::unordered_set<mgp::Value> set;
 
-    for (const auto &elem: list1){
+    for (const auto &elem : list1) {
```

Comment:
I think with new Memgraph this can actually be changed to work as was intended:

`std::unordered_set<mgp::Value> set(list1.begin(), list2.begin());`

Reply:
This should be fixed with Idora's PR on Memgraph with C++ API traits added under Iterator class in List